web/src/settings_bots.ts

```diff
@@ -190,32 +217,38 @@ function update_add_bot_button() {
     }
 }
 
-export function update_bot_permissions_ui() {
+export function update_bot_permissions_ui(): void {
     update_bot_settings_tip($("#admin-bot-settings-tip"), true);
     update_bot_settings_tip($("#personal-bot-settings-tip"), false);
     update_add_bot_button();
     $("#id_realm_bot_creation_policy").val(realm.realm_bot_creation_policy);
 }
 
-export function add_a_new_bot() {
+export function add_a_new_bot(): void {
     const html_body = render_add_new_bot_form({
         bot_types: page_params.bot_types,
         realm_embedded_bots: realm.realm_embedded_bots,
         realm_bot_domain: realm.realm_bot_domain,
     });
 
-    let create_avatar_widget;
+    let create_avatar_widget: UploadWidget;
 
-    function create_a_new_bot() {
-        const bot_type = $("#create_bot_type").val();
-        const full_name = $("#create_bot_name").val();
-        const short_name = $("#create_bot_short_name").val() || $("#create_bot_short_name").text();
+    function create_a_new_bot(): void {
+        const bot_type = $<HTMLSelectOneElement>("select:not([multiple])#create_bot_type").val()!;
+        const full_name = $<HTMLInputElement>("input#create_bot_name").val()!;
+        const short_name =
+            $<HTMLInputElement>("input#create_bot_short_name").val() ??
+            $<HTMLInputElement>("input#create_bot_short_name").text();
```

Comment:
`.text()` is not used for `input` elements. For `input` elements `.val()` is used. 

> The .text() method cannot be used on form inputs or scripts. To set or get the text value of input or textarea elements, use the [.val()](https://api.jquery.com/val/) method. To get the value of a script element, use the [.html()](https://api.jquery.com/html/) method.

https://api.jquery.com/text/#text

Hence 
```typescript
$<HTMLInputElement>("input#create_bot_short_name").text();
```
does not make much sense.